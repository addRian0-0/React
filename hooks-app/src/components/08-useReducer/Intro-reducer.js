//Nomas lo vi jejeje pero no lo hice, pero si le entendi