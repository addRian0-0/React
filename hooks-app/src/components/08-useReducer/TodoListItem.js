import React from 'react'

export const TodoListItem = ({ todo, handleDelete, handleTodo, index }) => {
    return (
        <div>
            <li
                key={todo.id}
                className="list-group-item"
            >
                <p
                    className={`${todo.done && 'complete'}`}
                    onClick={() => handleTodo(todo.id)}
                >
                    {index + 1}. {todo.desc}
                </p>
                <button
                    className="btn btn-danger"
                    onClick={() => { handleDelete(todo.id) }}
                >
                    Borrar
                </button>
            </li>
        </div>
    )
}
