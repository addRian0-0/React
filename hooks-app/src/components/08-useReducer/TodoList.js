import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({ todos, handleDelete, handleTodo }) => {
    return (
        <div>
            <ul className="list-group list-group-flush">
                {
                    todos.map((todo, i) => {
                        return <TodoListItem
                            todo={todo}
                            key={todo.id}
                            handleDelete={handleDelete}
                            handleTodo={handleTodo}
                            index={i}
                        />
                    })
                }
            </ul>
        </div>
    )
}
