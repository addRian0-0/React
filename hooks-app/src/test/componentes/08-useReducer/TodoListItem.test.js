import { shallow } from "enzyme";
import { TodoListItem } from "../../../components/08-useReducer/TodoListItem";
import { demosTodo } from "../../fixtures/demoTodos";

describe('pruebas en un elemento de la lista', () => {

    const handleDelete = jest.fn();
    const handleTodo = jest.fn();

    const wrapper = shallow(
        <TodoListItem
            todo={demosTodo[0]}
            handleDelete={handleDelete}
            handleTodo={handleTodo}
            index={1}
        />
    );

    test('hacer match con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe jalar handletodo', () => {
        
        wrapper.find('p').simulate('click');
        expect(handleTodo).toHaveBeenCalledWith(demosTodo[0].id);

    })

    test('debe mostrar el texto correctamente', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(`2. ${demosTodo[0].desc}`)

    })   

})
