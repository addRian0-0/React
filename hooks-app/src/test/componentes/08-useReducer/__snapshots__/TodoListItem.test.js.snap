// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en un elemento de la lista hacer match con el snapshot 1`] = `
<div>
  <li
    className="list-group-item"
    key="1"
  >
    <p
      className="undefined"
      onClick={[Function]}
    >
      2
      . 
      Aprender fastify
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Borrar
    </button>
  </li>
</div>
`;
